### ISSUE TITLE
Impact between Device and Network Card cause issues

### ISSUE THREAD

**Issue by ajasa19** [tomolimo/impacts#10]
If you have a device and try to have that device impact or being impacted by a network card all other devices linked to said device will disappear from the impact tables.

Also when having an impact between the two objects the topology diagram will display the device just fine but will not display text or an image for network cards only an arrow pointing to nothing.

![Screenshot from 2020-02-14 08-42-11](https://user-images.githubusercontent.com/61054231/74562692-58d05580-4f39-11ea-9131-b7294eeeb888.png)

**Comment by tomolimo** [comment]
Hello @ajasa19, 
Sorry for being late to answer, but I was out of office for illness.

Could you detail the steps in order for me to be able to reproduce the issue?

Thank you,
Regards,
Tomolimo

**Comment by ajasa19** [comment]
Sorry of the late reply had some stuff going on. 

So I have 1 network device named "NetDevice01" along with a component network card called "NetworkCard01" when trying to create an impact between these two items the diagram breaks along with the list of impacts.

 I have fixed the invisible image issue by adding a png file in the pics directory called "DeviceNetworkCard.png".

but the name of the network device won't show when in the topology diagram it sees the network card can't get a label as shown by it's node string value 

{ id: 'DeviceNetworkCard[3]', label: '', image: '/glpi/plugins/impacts/pics//DeviceNetworkCard.png', shape: 'image', urllink: '/glpi/front/device.form.php?itemtype=DeviceNetworkCard&id=3' }

I'm currently trying to figure out how it grabs labels for other devices.

**Comment by tomolimo** [comment]
Hello @ajasa19 
I've reproduced this issue, and the root cause is the fact that this king of assets doesn't have a name field...
I'll provide something else.
Thank you
Regards,
Tomolimo

**Comment by ajasa19** [comment]
I was able to fix my problem, It seems devices don't use the column "name" in the glpidb they use "item type".

so what I did was I made an if statement where if the item type was associated to device network cards it word call on a different way to get $nodestring.

```
if ($data['itemtype'] == 'DeviceNetworkCard'){
	$nodestring .= "{ id: '$id', label: '". $temp->fields['designation']."', image: '$imgpath/".$data['itemtype'].".png', shape: 'image', urllink: '$link' $addOptions},";
} else {
         $nodestring .= "{ id: '$id', label: '".$temp->fields['name']."', image: '$imgpath/".$data['itemtype'].".png', shape: 'image', urllink: '$link' $addOptions},";
}
```
this only works with Device Network Cards only though devices such as batteries will still have no name.

Hopefully this helps with the next update of impact.

**Comment by tomolimo** [comment]
Hello,
this issue has been fixed by PR #11 
Regards
Tomolimo
